src/admin.cc

```diff
@@ -491,11 +491,26 @@ class AdminServiceImpl final : public zsearch::AdminService::Service {
                                     std::mutex& out_mtx,
                                     uint32_t max_records,
                                     std::atomic<std::uint32_t>& count) {
+        std::unique_ptr<DeltaHandler> delta_handler =
+                m_delta_ctx->new_certificate_delta_handler();
         for (auto it = cert_store.seek_start_of_shard(shard_id);
              it.valid() && cert_store.shard_for(it->first) == shard_id; ++it) {
             if (max_records && ++count > max_records) {
                 return;
             }
+            // Only update expiration if `not_valid_before` and
+            // `not_valid_after` are actually set.
+            if (it->second.certificate().not_valid_before() &&
+                it->second.certificate().not_valid_after()) {
+                std::time_t now = std::time(nullptr);
+                bool expired =
+                        !certificate_valid_at(it->second.certificate(), now);
+                if (expired != it->second.certificate().expired()) {
+                    it->second.mutable_certificate()->set_expired(expired);
+                    AnonymousResult res = cert_store.put(it->second);
+                    delta_handler->handle_delta(res);
```

Comment:
We also need to set `valid` and `trusted_path` based on what's in `was_valid` and `was_trusted_path`.